pandas/core/arrays/datetimes.py

```diff
@@ -324,6 +430,26 @@ def _generate_range(cls, start, end, periods, freq, tz=None,
 
         return cls._simple_new(index.asi8, freq=freq, tz=tz)
 
+    # -----------------------------------------------------------------
+    # DatetimeLike Interface
+
+    def _unbox_scalar(self, value):
+        if not isinstance(value, (self._scalar_type, type(NaT))):
```

Comment:
see my comment above about NaT

Reply:
Your comment referencing adding NaT to `scalar_types`?

I think I had something like `if not isinstance(values, self.scalar_types`) originally, but that isn't quite right for datetime. We need to be specific in each subclass. If `value` were a Period or Timestamp here, we would want it to raise, since this is for datetime.